osbenchmark/workload_generator/corpus.py

```diff
@@ -62,24 +62,27 @@ def extract(client, output_path, index):
 
     logger = logging.getLogger(__name__)
 
-    total_docs = client.count(index=index)["count"]
-    if total_docs > 0:
-        logger.info("[%d] total docs in index [%s].", total_docs, index)
+    number_of_docs = client.count(index=index)["count"]
+
+    ndocs_to_extract = number_of_docs if not number_of_docs_requested else min(number_of_docs, number_of_docs_requested)
+
+    if ndocs_to_extract > 0:
+        logger.info("[%d] total docs in index [%s].", number_of_docs, index)
         docs_path = get_doc_outpath(output_path, index)
-        dump_documents(client, index, get_doc_outpath(output_path, index, "-1k"), min(total_docs, 1000), " for test mode")
-        dump_documents(client, index, docs_path, total_docs)
-        return template_vars(index, docs_path, total_docs)
+        dump_documents(client, index, get_doc_outpath(output_path, index, "-1k"), min(ndocs_to_extract, 1000), " for test mode")
```

Comment:
IMO, the doc count specified should not be respected for test mode -- it should always be min(1000, total_docs).  Users generally don't know the internals of test mode and it is unlikely they intend to specify that value.